### DISCUSSION TITLE
Cyclic data.

### DISCUSSION THREAD

**Discussion by anubhavkamal** [MilesCranmer/PySR#567] [Q&A]
Hello @MilesCranmer ,
I am working on cyclic data of the form -

```
data = {"1" : {"time" : [1,2,3,4,5],"volt" : [1,2,3,4,5], "current":[1.1,1.4,1.3,1.5,1.5], "capacity": [0.5, 0.6, 0.7, 0.8, 0.9]},
"2" : {"time" : [1,2,3,4,5],"volt" : [1.1,2.1,3.5,4.2,1.7], "current":[0.1,0.4,0.3,0.5,0.5], "capacity": [0.2, 0.3, 0.4, 0.5, 0.5]}
...}
```
The data is made up to showcase the structure of dataset.
Here, "1" and "2" are cycle numbers respectively. For each cycle say "1", we have behavior of the system i.e. current, voltage etc.
I have used SR independently for each cycle and I am able to get decent accuracy (target = capacity).
Is it possible to make it a function of cycle number as well?
How can I parse this type of data to PySR?
Thanks!

**Comment by MilesCranmer** [comment]
Can you simply introduce a new feature for the cycle number? Then you would have 10 rows - half of them with cycle=1 and half with cycle=2.

**Reply by anubhavkamal** [reply]
Hello @MilesCranmer !
Thank you for your reply. 
I tried this but it doesn't seem to generalise that well.
In some cycles the equation involves exponential function and in others it might find a solution with sin function in it.
The dataset is fairly big as compared to example I have shown here.
I am open to suggestions involving multiple regressors for each cycle then combine them somehow?
Or something similar to change the form of dataset itself.
Did not having any luck yet.
Thanks!
